crates/bevy_render/src/mesh/shape/torus.rs

```diff
@@ -0,0 +1,96 @@
+use crate::{
+    mesh::{Indices, Mesh},
+    pipeline::PrimitiveTopology,
+};
+use bevy_math::{Quat, Vec3};
+
+/// A torus (donut) shape.
+#[derive(Debug)]
+pub struct Torus {
+    pub radius: f32,
+    pub tube_radius: f32,
+    pub subdivisions_segments: usize,
+    pub subdivisions_sides: usize,
+}
+
+impl Default for Torus {
+    fn default() -> Self {
+        Torus {
+            radius: 1.0,
+            tube_radius: 0.5,
+            subdivisions_segments: 32,
+            subdivisions_sides: 24,
+        }
+    }
+}
+
+impl From<Torus> for Mesh {
+    fn from(torus: Torus) -> Self {
+        // code adapted from http://wiki.unity3d.com/index.php/ProceduralPrimitives#C.23_-_Torus
```

Comment:
Unity Wiki content is Creative Commons ShareAlike 3.0. This probably qualifies as a "remix", which means this code also must be released under that license.

We can _probably_ deal with a ShareAlike license, but its suboptimal because any changes made to this code must also be ShareAlike, and we need to label this file as such. In general I'd prefer it if we used content that is compatible with MIT to keep the licensing situation clear.

Reply:
Right, I understand. I'll look for another implementation that's more permissively licensed. MIT and CC-0 should be okay, right?

Reply:
Yup CC-0 is fine. It doesn't require attribution, but if we find code, lets provide attribution anyway (if we can).